File: trunk/wasabi/devices/unicone/led_test.py
#!/usr/bin/env python
import pyunicone, time, sys


class ProgressOperation:
    def __init__(self, name):
        sys.stderr.write("Starting %s: " % name)

    def update(self, completion):
        sys.stderr.write("%.02f " % completion)

    def finish(self, message):
        if message:
	    sys.stderr.write("[%s]\n" % message)
	else:
            sys.stderr.write("Done.\n")

class MyProgressReporter:
    def start(self, op):
        return ProgressOperation(op)


dev = pyunicone.UniconeDevice()
dev.progress = MyProgressReporter()
dev.configure("fpga/test/test.bit")

while 1:
    dev.setLed(1, 0.015)
    time.sleep(0.5)

File: trunk/wasabi/devices/unicone/src/led_test.c
/*
 * Demonstrate LED brightness control
 */

#include <unicone.h>
#include "unicone_device.h"
#include "progress.h"

int main()
{
  struct unicone_device* dev;
  struct progress_reporter* progress = progress_reporter_console_new();

  unicone_usb_init();
  dev = unicone_device_new();
  if (!dev) {
    printf("Can't open unicone device\n");
    return 1;
  }

  if (unicone_device_configure(dev, "firmware.bin", "fpga/test/test.bit", progress) < 0) {
    printf("Error configuring unicone device\n");
    return 1;
  }

  while (1) {
    unicone_device_set_led(dev, 1, 0.015);
    sleep(1);
  }

  return 0;
}

